controllers/awsmachine_controller.go

```diff
@@ -911,3 +918,32 @@ func (r *AWSMachineReconciler) indexAWSMachineByInstanceID(o client.Object) []st
 
 	return nil
 }
+
+func (r *AWSMachineReconciler) ensureStorageTags(ec2svc services.EC2MachineInterface, instanceID *string, machine *infrav1.AWSMachine) error {
+	volumes, err := ec2svc.GetEC2Volumes(instanceID)
+	if err != nil {
+		return err
+	}
+
+	for _, ebsVolume := range volumes {
+		volume := sdkToVolume(ebsVolume)
+		_, err = r.ensureVolumeTags(ec2svc, machine, volume, instanceID, machine.Spec.Tags)
```

Comment:
We only need to pass in a volume ID. Don't need a full `infrav1.Volume` in the signature. 
We only want to apply the existing AdditionalTags in the API to the volume, not apply a different set of tags.